src/charm.py

```diff
@@ -136,13 +156,90 @@ def _update_layer(self) -> None:
 
+    def _get_manifests(self, interfaces, relation, event):
+        """Unpacks and returns the manifests relation data."""
+        if not ((relation_interface := interfaces[relation]) and relation_interface.get_data()):
+            logging.info(f"No {relation} data presented in relation")
+            return None
+        try:
+            relations_data = {
+                (rel, app): route
+                for (rel, app), route in sorted(
+                    relation_interface.get_data().items(), key=lambda tup: tup[0][0].id
+                )
+                if app != self.app
+            }
+        except Exception as e:
+            raise ErrorWithStatus(
+                f"Unexpected error unpacking {relation} data - data format not "
+                f"as expected. Caught exception: '{str(e)}'",
+                BlockedStatus,
+            )
+        if isinstance(event, (RelationBrokenEvent)):
+            del relations_data[(event.relation, event.app)]
+
+        manifests = []
+        for relation_data in relations_data.values():
+            manifests += json.loads(relation_data[relation])
+        self.logger.info(f"manifests are {manifests}")
+        return manifests
+
+    def _manifests_valid(self, manifests):
+        """Checks if manifests are unique."""
+        if manifests:
+            for manifest in manifests:
+                if (
+                    sum([m["metadata"]["name"] == manifest["metadata"]["name"] for m in manifests])
+                    > 1
+                ):
+                    return False
+        return True
+
+    def _sync_manifests(self, manifests, push_location):
+        """Push list of manifests into layer.
+
+        Args:
+            manifests: List of kubernetes manifests to be pushed to pebble layer.
+            push_location: Container location where the manifests should be pushed to.
+        """
+        all_files = self.container.list_files(push_location)
+        if manifests:
+            manifests_locations = [
+                f"{push_location}/{m['metadata']['name']}.yaml" for m in manifests
+            ]
+        else:
+            manifests_locations = []
+        if all_files:
+            for file in all_files:
+                if file.path not in manifests_locations:
+                    self.container.remove_path(file.path)
+        if manifests:
+            for manifest in manifests:
+                filename = manifest["metadata"]["name"]
+                self.container.push(f"{push_location}/{filename}.yaml", yaml.dump(manifest))
+
+    def _update_manifests(self, interfaces, dispatch_folder, relation, event):
+        """Get manifests from relation and update them in dispatcher folder."""
+        manifests = self._get_manifests(interfaces, relation, event)
+        if not self._manifests_valid(manifests):
+            self.logger.debug(
+                f"Manifests names in all relations must be unique, received manifests {manifests}"
```

Comment:
is manifests a list?
```suggestion
                f"Manifests names in all relations must be unique, received manifests { ','.join(manifests)}"
```